### ISSUE TITLE
OTLP --> Jaeger trace transformation leads to missing properties

### ISSUE THREAD

**Issue by secustor** [open-telemetry/opentelemetry-collector#815]
Version: 0e1b2e323d398cf6cda500953c7896aa01c173b7       
Instrumentation: https://github.com/open-telemetry/opentelemetry-auto-instr-java v0.2.2
Logs: https://gist.github.com/secustor/1d31b3ac613208fef7d6e50912528f07

I use following setup:    
`[opentelemetry-auto-instr-java] --otlp--> [OTEL collector] --jaeger_grpc--> [jaeger all in one]`
this leads to `"process is empty for the span"` errors jaeger as the collector seems not set resource attributes like process informations. 

A direct setup is working fine: 
`[opentelemetry-auto-instr-java] --jaeger_grpc--> [jaeger all in one]`

**Comment by RashmiRam** [comment]
@secustor I faced the same error. I tried adding env variable **OTEL_RESOURCE_ATTRIBUTES** That solved the issue. 

Example:

`OTEL_RESOURCE_ATTRIBUTES=service.name=<service_name>,host.name=<host_name>,host.ip=<host_ip> java -javaagent:path/to/opentelemetry-auto-0.2.2.jar  -Dota.exporter.jar=/path/to/opentelemetry-auto-exporters-otlp-0.2.2.jar  -Dota.exporter.otlp.endpoint=localhost:9090 -jar my_jar.jar`

However in the jaeger UI, the traces are presented with <trace-without-root-span>

**Comment by bogdandrutu** [comment]
@trask can you help with this, it seems to be related to auto instrumentation.

**Comment by sleighzy** [comment]
@RashmiRam , the trace without root span problem is possibly due to this issue, https://github.com/open-telemetry/opentelemetry-java/issues/1164. I had experienced something similar when viewing my traces in Jaeger, when using the OTel Collector, which looked to be due to the root parent spans being dropped due their default parent span ids being all zeros.

**Comment by trask** [comment]
Thanks @sleighzy! It looks like that fix just snuck into the opentelemetry-java 0.4.0 release, so it should be in the latest auto-instr-java snapshots already: https://app.circleci.com/pipelines/github/open-telemetry/opentelemetry-auto-instr-java/982/workflows/286e781c-a2d4-4068-b567-b727521fdf78/jobs/8138/artifacts

**Comment by stefanprisca** [comment]
I am seeing the same issue from the go sdk, and the go otlp exporter does [parent name validation](https://github.com/open-telemetry/opentelemetry-go/blob/master/exporters/otlp/internal/transform/span.go#L89). When I connect directly to jaeger it works (as in [this](https://github.com/open-telemetry/opentelemetry-go/blob/master/example/jaeger/main.go) example). But when sending the traces through the collector, Jaeger starts complaining that "process is empty for the span". 
I'm using the simplest Jaeger set-up from microk8s and [this](https://github.com/stefanprisca/opentelemetry-go/blob/otel-collector-demo/example/otel-collector/otel-coll.yaml) configuration for the otel-collector.

**Comment by sleighzy** [comment]
@stefanprisca , I don't know enough about Go, but to resolve that "process is empty for the span" with the Java OTLP exporter I had to set the `OTEL_RESOURCE_ATTRIBUTES` as per the comment by @RashmiRam , see https://github.com/open-telemetry/opentelemetry-collector/issues/815#issuecomment-618860026. Without this the Resource attributes aren't created when the exporter is started, which are used by Jaeger to populate the Process tags and displayed in the Jaeger UI. This is a different problem than the "parent name validation" code link you provided.

**Comment by sleighzy** [comment]
@stefanprisca , based on your code link for the example you said works it has this, https://github.com/open-telemetry/opentelemetry-go/blob/master/example/jaeger/main.go#L36, which is adding those process attributes. This may not be happening in your other usecase, e.g. the service name is not being added as a Resource attribute when the exporter is created.

**Comment by stefanprisca** [comment]
@sleighzy I see, thanks for the clarifications! There must be something I'm missing on the Go side.

**Comment by damemi** [comment]
I was seeing a similar issue as @stefanprisca (discussed in #880) with an older version of the collector and the go library. The problem was that when I was using the OTLP exporter, I couldn't find a way to set the service name.

As mentioned in https://github.com/open-telemetry/opentelemetry-collector/issues/880#issuecomment-621499695, the `OCProtoToJaegerThrift` translator now fills in `unknown-service-name` automatically when translating OTLP traces to Jaeger, but is there a way to set this service name field when initializing the OTLP exporter? The Jaeger exporter has service name attributes (as mentioned in the example shown) but afaict the OTLP one does not

**Comment by stefanprisca** [comment]
@damemi  I've just spent more time looking into this, and there actually is a nice way of setting the service name from the go sdk. You have to do it by using `WithResourceAttributes` when creating the trace provider:
https://github.com/open-telemetry/opentelemetry-go/blob/88100f0fc752e57cc52e16ef2b4f3d520f0abc74/sdk/trace/provider.go#L200

Just make sure to call it after calling `WithConfig`, otherwise it will get overwritten. For example:
```go
tp, err := sdktrace.NewProvider(
		sdktrace.WithConfig(sdktrace.Config{DefaultSampler: sdktrace.AlwaysSample()}),
		sdktrace.WithResourceAttributes(
			core.Key(conventions.AttributeServiceName).String("test-service"),
		),
		sdktrace.WithBatcher(exp, // add following two options to ensure flush
			sdktrace.WithScheduleDelayMillis(5),
			sdktrace.WithMaxExportBatchSize(2),
		))
```
This works with the official 0.3.0 release for the collector. 
I'm working now on a more complete example for the go sdk (PR [711](https://github.com/open-telemetry/opentelemetry-go/pull/711))

**Comment by damemi** [comment]
@stefanprisca thanks, I didn't realize that. And the updated example you're working on looks helpful too!

**Comment by kbrockhoff** [comment]
Fix of this ticket needs to be coordinated with fixes made for #1137
